Sources/NIOHTTP1/ByteCollectionUtils.swift

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+fileprivate let defaultWhitespaces = [" ", "\t"].map({$0.utf8.first!})
+
+extension ByteBufferView {
+    internal func trim(limitingElements: [UInt8]) -> ByteBufferView {
+        let lastNonWhitespaceIndex = self.lastIndex { !limitingElements.contains($0) }
+        let firstNonWhitespaceIndex = self.firstIndex { !limitingElements.contains($0) }
+
+        return self[firstNonWhitespaceIndex..<lastNonWhitespaceIndex]
```

Comment:
I think this slice is wrong: specifically, I think it's off by one. `lastNonWhitespaceIndex` is the last index that does not contain whitespace, so we want to include that in our slice. I think this should be `self.index(after: lastNonWhitespaceIndex)`.

Reply:
Sorry, I don't have Swift 4.2 so I couldn't test,  so copied the source of firstIndex and lastIndex from the Swift source code

```swift
extension BidirectionalCollection {
    @inline(__always)
    public func firstIndex(
        where predicate: (Element) throws -> Bool
        ) rethrows -> Index? {
        var i = self.startIndex
        while i != self.endIndex {
            if try predicate(self[i]) {
                return i
            }
            self.formIndex(after: &i)
        }
        return nil
    }
    
    @inline(__always)
    public func lastIndex(
        where predicate: (Element) throws -> Bool
        ) rethrows -> Index? {
        var i = endIndex
        while i != startIndex {
            formIndex(before: &i)
            if try predicate(self[i]) {
                return i
            }
        }
        return nil
    }
}

```
Source: https://github.com/apple/swift/blob/33c2efe32e34c12679cfc135de4715c0cb905a71/stdlib/public/core/CollectionAlgorithms.swift

And found that:
  1. You are right,  it needs `self.index(after: lastNonWhitespaceIndex)`
  2. We also need to unwrap the return, because it is optional, it returns `Index?`